File: 0001. Two Sum.py
# 9/26/2023 solution
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        for i, num in enumerate(nums):
            compliment = target - num
            if compliment in dict:
                return [i, dict[compliment]]
            dict[num] = i
        return

# Previous solution
# class Solution:
#     def twoSum(self, nums: List[int], target: int) -> List[int]:
#         searched = {}
#         for i in range(len(nums)):
#             complement = target - nums[i]
#             if complement in searched:
#                 return [i, searched[complement]]
#             searched[nums[i]] = i

File: 0001. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) { 
        map<int, int> dict;

        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];
            
            if (dict.count(complement)) {
                return {dict[complement], i};
            }

            dict[nums[i]] = i;
        }

        return {};
    }
};
